google/cloud/aiplatform/metadata/artifact.py

```diff
@@ -0,0 +1,285 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import proto
+import logging
+from typing import Optional, Dict
+
+from google.api_core import exceptions
+from google.cloud.aiplatform import utils
+from google.cloud.aiplatform.metadata.resource import _Resource
+from google.auth import credentials as auth_credentials
+
+from google.cloud.aiplatform_v1beta1.types import artifact as gca_artifact
+
+
+class _Artifact(_Resource):
+    """Metadata Artifact resource for AI Platform"""
+
+    _resource_noun = "artifacts"
+    _getter_method = "get_artifact"
+
+    def __init__(
+        self,
+        artifact_name: str,
+        metadata_store_id: Optional[str] = "default",
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        """Retrieves an existing Artifact given an Artifact name or ID.
+
+        Args:
+            artifact_name (str):
+                A fully-qualified Artifact resource name or artifact ID
+                Example: "projects/123/locations/us-central1/metadataStores/default/artifacts/my-artifact".
+                or "my-artifact" when project and location are initialized or passed.
+            metadata_store_id (str):
+                MetadataStore to retrieve resource from. If not set, metadata_store_id is set to "default".
+                If artifact_name is a fully-qualified Artifact, its metadata_store_id overrides this one.
+            project (str):
+                Optional project to retrieve resource from. If not set, project
+                set in aiplatform.init will be used.
+            location (str):
+                Optional location to retrieve resource from. If not set, location
+                set in aiplatform.init will be used.
+            credentials (auth_credentials.Credentials):
+                Custom credentials to use to upload this model. Overrides
+                credentials set in aiplatform.init.
+        """
+
+        super().__init__(
+            resource_name=artifact_name,
+            metadata_store_id=metadata_store_id,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+    @classmethod
+    def create(
+        cls,
+        artifact_id: str,
+        schema_title: str,
+        display_name: Optional[str] = None,
+        schema_version: Optional[str] = None,
+        description: Optional[str] = None,
+        metadata: Optional[Dict] = {},
```

Comment:
This should be set to `None` since the default here is mutable.

Reply:
Done, thanks for the pointer!